### PR TITLE
[RSDK-5643] return raw readings when reading analog values

### PR DESCRIPTION
**Pull request by npmenard**

### CODE REVIEW

File: src/esp32/analog.rs [hunk 1]
```diff
@@ -19,22 +23,36 @@ where
...
+        self.raw_reading()
             .map_err(|e| anyhow::anyhow!(format!("error while reading analog reader {e}")))
     }
     fn inner_name(&self) -> String {
         self.name.clone()
     }
+    fn raw_reading(&mut self) -> anyhow::Result<u16> {
+        let unit = T::Adc::unit();
+        let channel = self.pin.0;
+        if unit == adc_unit_t_ADC_UNIT_1 {
```
Comment (acmorrow) [inline src/esp32/analog.rs:45]: Maybe invert this conditional to handle the error case first:

```
if unit != adc_unit ... {
    return Err(...)
}

let measurement = ...
```

File: src/esp32/analog.rs [hunk 2]
```diff
@@ -19,22 +23,36 @@ where
...
+            let measurement = unsafe { adc1_get_raw(channel) };
```
Comment (acmorrow) [inline src/esp32/analog.rs:46]: Why doesn't `AdcDriver::read_raw` do what you need here?

**Review by acmorrow — COMMENTED** [review]
In my own words, my best understanding of what is happening here is that we used to just use `AdcDriver::read` which returned a reading in millivolts against calibration settings, but per the ticket we really just want the raw reading to be consistent with RDK, so dropping down to `adc1_get_raw` directly. But I'm curious why `AdcDriver::read_raw` couldn't be used, as it seems to arrive at `adc1_get_raw` anyway: https://esp-rs.github.io/esp-idf-hal/src/esp_idf_hal/adc.rs.html#298.

**Comment by npmenard** [comment]
> In my own words, my best understanding of what is happening here is that we used to just use `AdcDriver::read` which returned a reading in millivolts against calibration settings, but per the ticket we really just want the raw reading to be consistent with RDK, so dropping down to `adc1_get_raw` directly. But I'm curious why `AdcDriver::read_raw` couldn't be used, as it seems to arrive at `adc1_get_raw` anyway: https://esp-rs.github.io/esp-idf-hal/src/esp_idf_hal/adc.rs.html#298.
> 
> 

You're right I missed the function altogether -_-

File: src/esp32/analog.rs (hunk 2 above)
Comment (npmenard) [reply src/esp32/analog.rs:46]: I withdraw my comment, we would need to update esp-idf-hal which will need us to update to rust 1.71 which snowballs into an image that will not fit anymore in the 4MB address space

Comment (acmorrow) [reply src/esp32/analog.rs:46]: Ah, that's unfortunate. Maybe leave a comment in the hopes that we can revisit later?

**Review by acmorrow — APPROVED** [review]

**Review by acmorrow — APPROVED** [review]

**Pull request merged**